kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -663,6 +683,97 @@ define([
+                for (const [rowId, value] of Object.entries(outputValues)) {
+                    if (!value) {
+                        return counts;
+                    } else if (!(value in counts)) {
+                        counts[value] = [];
+                    }
+                    counts[value].push({ rowId, paramId });
+                }
+                return counts;
+            }, {});
+            const values = Object.keys(outputValueCounts).reduce((duplicateValues, value) => {
+                if (outputValueCounts[value].length > 1) {
+                    duplicateValues[value] = outputValueCounts[value];
+                }
+                return duplicateValues;
+            }, {});
+            // values should now look like:
+            // {
+            //   'field_value': [{ rowId1, paramId }, { rowId2, paramId }]
+            // }
+            // with only entries where there are at least two row ids
+            // translate to something easier to work with -
+            // {
+            //    rowId: {
+            //      paramId1: {
+            //        widget: Object
+            //        duplicateRows: [rowId, rowId]
+            //      },
+            //      paramId2: {
+            //        widget: Object,
+            //        duplicateRows: [rowId, rowId]
+            //    }
+            // }
+            const duplicates = {};
+            Object.values(values).forEach((dupArr) => {
+                const allRows = dupArr.reduce((rowIds, entry) => {
+                    rowIds.add(dataModel.rowIdToIndex[entry.rowId] + 1);
+                    return rowIds;
+                }, new Set());
+                dupArr.forEach((entry) => {
+                    const rowId = entry.rowId;
+                    const rowIdx = dataModel.rowIdToIndex[rowId] + 1;
+                    const paramInfo = {
+                        widget: dataModel.rows[rowId].widgets[entry.paramId],
+                        duplicateRows: [...allRows].filter((x) => x !== rowIdx),
+                    };
+                    if (paramInfo.duplicateRows.length === 0) {
+                        paramInfo.duplicateRows.push(rowIdx);
+                    }
+                    if (!(rowId in duplicates)) {
+                        duplicates[rowId] = {};
+                    }
+                    duplicates[rowId][entry.paramId] = paramInfo;
+                });
+            });
+            const outputParams = new Set(getAllOutputParameterWidgets());
+
+            Object.values(duplicates).forEach((duplicateEntry) => {
+                Object.values(duplicateEntry).forEach((entry) => {
+                    entry.widget.setDuplicateValue(entry.duplicateRows);
+                    outputParams.delete(entry.widget);
+                });
+            });
+            outputParams.forEach((widget) => widget.clearDuplicateValue());
+
+            return Object.keys(values);
```

Comment:
The calling function gets the Array of values that were duplicated. Currently only used to respond to the `changed` message if the changed value is in that Array, then we need to pass an error message up to the parent widget.